### PR TITLE
WIP: dev/eahw-2109/publish-jparest-artifactory: code to publish jparest li…

### PR DESCRIPTION
**Pull request by JWeston-HO**
publish the jparest library to artifactory

### CODE REVIEW

File: jparest_settings.xml [hunk 1]
```diff
@@ -0,0 +1 @@
...
                <httpHeaders>
                    <property>
                        <name>Authorization</name>
                        <value>Bearer ${ARTIFACTORY_TOKEN}</value>
                    </property>
                </httpHeaders>
            </configuration>
        </server>
    </servers>
</settings>
```
Comment (laurieboyes) [inline jparest_settings.xml:1]: Probably not important but from a github point of view at least it’d be nice to fix these carriage returns 😄

File: .drone.yml [hunk 2]
```diff
@@ -14,27 +14,35 @@ trigger:
...
-  - "BRANCH=$(echo $DRONE_BRANCH | tr '[:lower:]' '[:upper:]') && [[ $BRANCH =~ (EAHW-[0-9]*) ]] && projectSuffix=$${match/#E/-E}$${BASH_REMATCH/#E/-E}"
-  - mvn sonar:sonar -Dsonar.host.url=$${SONAR_HOST} -Dsonar.login=$${SONAR_TOKEN} -DprojectKeySuffix=$${projectSuffix}
-  when:
-    event:
-      exclude:
-      - pull_request
+  - name: sonar
+    image: maven:3.8-openjdk-17
+    depends_on:
+      - publish
```
Comment (laurieboyes) [inline .drone.yml:33]: So sonarcube runs _after_ the lib is published? 🤔

File: jparest_settings.xml (hunk 1 above)
Comment (JWeston-HO) [reply jparest_settings.xml]: fixed

**Review by laurieboyes — DISMISSED** [review]

File: .drone.yml [hunk 3]
```diff
@@ -36,6 +36,21 @@ steps:
...
+  environment:
+    ARTIFACTORY_TOKEN:
+      from_secret: artifactory_token
+  commands:
+  - mvn clean install
+  - mvn -s ./jparest_settings.xml -f pom.xml install -B org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy -DuniqueVersion=false -DaltReleaseDeploymentRepository=artifactory::default::https://artifactory.digital.homeoffice.gov.uk/artifactory/callisto-jparest -DskipTests
+  when: 
+    event:
+     exclude:
+     - pull_request
```
Comment (stephenkaye-ho) [inline .drone.yml:52]: Doesn't this mean that we are publishing on every push to any branch without sorting out the versioning

File: .drone.yml [hunk 4]
```diff
@@ -36,6 +36,21 @@ steps:
...
+- name: publish
+  image: maven:3.8-openjdk-17
+  depends_on:
+  - sonar
+  environment:
+    ARTIFACTORY_TOKEN:
+      from_secret: artifactory_token
+  commands:
+  - mvn clean install
+  - mvn -s ./jparest_settings.xml -f pom.xml install -B org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy -DuniqueVersion=false -DaltReleaseDeploymentRepository=artifactory::default::https://artifactory.digital.homeoffice.gov.uk/artifactory/callisto-jparest -DskipTests
```
Comment (stephenkaye-ho) [inline .drone.yml:48]: Is this needed altReleaseDeploymentRepository?
If the id in the jparest_settings.xml was set to match the repositories in the pom.xml changes then I don't think it's required

File: .drone.yml (hunk 3 above)
Comment (JWeston-HO) [reply .drone.yml]: I have addressed this by limiting the push trigger to only trigger the web hook on the main branch.

File: .drone.yml (hunk 4 above)
Comment (JWeston-HO) [reply .drone.yml]: This is correct. I have removed it

**Pull request closed without merging**